cmake/extensions.cmake

```diff
@@ -33,6 +33,19 @@ function(add_overlay_config image overlay_file)
   endif()
 endfunction()
 
+# Add a DTS overlay file to a child image.
+# This can be used by a parent image to set DTS overlay in its child images
+# This function must be called before 'add_child_image(image)' to have effect.
+function(add_dtc_overlay image dtc_overlay_file)
+  set(old_dtc ${${image}_DTC_OVERLAY_FILE})
```

Comment:
This seems to duplicates the logic from `add_overlay_config_image`, if so: could you extract a macro and use that in both cases?

Reply:
Changed to single function. Added two macros that implicitly provide `DTC_OVERLAY_FILE` or `OVERLAY_CONFIG` as 3rd argument values.